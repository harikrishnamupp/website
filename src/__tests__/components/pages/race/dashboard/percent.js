import React from 'react'
import renderer from 'react-test-renderer'
import Percent from '~components/pages/race/dashboard/percent'

describe('Components : Race : Dashboard : Percent', () => {
  it('renders correctly', () => {
    const zeroTree = renderer.create(<Percent number={0} />).toJSON()
    expect(zeroTree).toMatchSnapshot()

    const nullTree = renderer.create(<Percent number={null} />).toJSON()
    expect(nullTree).toMatchSnapshot()

    const greaterThanOneTree = renderer
      .create(<Percent number={0.2} />)
      .toJSON()
    expect(greaterThanOneTree).toMatchSnapshot()

    const lessThanOneTree = renderer
      .create(<Percent number={0.00034} />)
      .toJSON()
    expect(lessThanOneTree).toMatchSnapshot()
  })
})
