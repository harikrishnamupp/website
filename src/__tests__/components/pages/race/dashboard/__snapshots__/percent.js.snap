// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Race : Dashboard : Percent renders correctly 1`] = `
Array [
  "0",
  "%",
]
`;

exports[`Components : Race : Dashboard : Percent renders correctly 2`] = `"-"`;

exports[`Components : Race : Dashboard : Percent renders correctly 3`] = `
Array [
  "20",
  "%",
]
`;

exports[`Components : Race : Dashboard : Percent renders correctly 4`] = `
Array [
  "<1",
  "%",
]
`;
